rubin_sim/maf/plots/skyproj_plotters.py

```diff
@@ -0,0 +1,523 @@
+            colorbar.solids.set_edgecolor("face")
+
+    def decorate(self):
+        super().decorate()
+
+        if self.plot_dict["colorbar"]:
+            self.draw_colorbar()
+
+    def _transform_to_healpix(self, metric_value_in, slicer):
+        # Bin the values up on a healpix grid.
+        metric_value = _healbin(
+            slicer.slice_points["ra"],
+            slicer.slice_points["dec"],
+            metric_value_in.filled(slicer.badval),
+            nside=self.plot_dict["nside"],
+            reduce_func=self.plot_dict["reduce_func"],
+            fill_val=slicer.badval,
+        )
+        mask = np.zeros(metric_value.size)
+        mask[np.where(metric_value == slicer.badval)] = 1
+
+        metric_value = ma.array(metric_value, mask=mask)
+        return metric_value
+
+    def _compatible_color_limits(self, norm, clims):
+        # Log can't be 0
+        compatible = (norm != "log") or (clims[0] < 0 and clims[1] < 0) or (clims[0] > 0 and clims[1] > 0)
+        return compatible
+
+    def draw(self, metric_values_in, slicer):
+        """Draw the healpix map.
+
+        Parameters
+        ----------
+        metric_value : `numpy.ma.MaskedArray`
+            The metric values from the bundle.
+        slicer : `rubin_sim.maf.slicers.TwoDSlicer`
+            The slicer
+        """
+        kwargs = {}
+        kwargs.update(self.default_hpixmap_kwargs)
+
+        # Check if we have a valid HEALpix slicer
+        if "Heal" not in slicer.slicer_name:
+            hpix_map = self._transform_to_healpix(metric_values_in, slicer)
+        else:
+            hpix_map = metric_values_in.copy()
+
+        hpix_map = apply_zp_norm(hpix_map, self.plot_dict)
+
+        if "draw_hpxmap_kwargs" in self.plot_dict:
+            kwargs.update(self.plot_dict["draw_hpxmap_kwargs"])
+
+        try:
+            kwargs["cmap"] = set_color_map(self.plot_dict)
+        except AttributeError:
+            # Probably here because an invalid cmap was set
+            pass
+
+        if self.plot_dict["log_scale"]:
+            if "norm" in kwargs and kwargs["norm"] != "log":
+                raise ValueError("Contradictory values for color scale normalization set.")
+            kwargs["norm"] = "log"
+
+        clims = set_color_lims(hpix_map, self.plot_dict)
+        if "norm" not in kwargs or self._compatible_color_limits(kwargs["norm"], clims):
+            kwargs["vmin"] = clims[0]
+            kwargs["vmax"] = clims[1]
+        else:
+            warnings.warn(
+                "Using log normalization, but color limits pass through 0. "
+                "Adjusting so plotting doesn't fail"
+            )
+
+        # Filling in masked values must take place after automatic color limits
+        # might be calculated!
+
+        # This masking replicates the behavior of HealpixSkyMap, but is it
+        # what we really want to do? Why not use the default handling of
+        # masked values applied by skyproj?
```

Comment:
I think here, "mask_below" was different than "a metric value which was masked". 
In general, I don't use "mask_below" and I think it was added by @yoachim but I am not sure if there is still a need for it?
I guess what you're saying here, is why not set the HealSparse 'sentinel values' to the mask_below limit, instead of just hp.UNSEEN?